core/dbt/task/unit_test.py

```diff
@@ -206,13 +207,25 @@ def build_unit_test_manifest(self):
         return loader.load()
 
     def reset_job_queue_and_manifest(self):
+        # We want deferral to happen here (earlier than normal) before we turn
+        # the normal manifest into the unit testing manifest
+        adapter = get_adapter(self.config)
+        with adapter.connection_named("master"):
+            self.populate_adapter_cache(adapter)
+            self.defer_to_manifest(adapter, self.job_queue._selected)
+
         # We have the selected models from the "regular" manifest, now we switch
         # to using the unit_test_manifest to run the unit tests.
         self.using_unit_test_manifest = True
         self.manifest = self.build_unit_test_manifest()
         self.compile_manifest()  # create the networkx graph
         self.job_queue = self.get_graph_queue()
 
+    def before_run(self, adapter, selected_uids: AbstractSet[str]) -> None:
+        # We already did cache population + deferral earlier (above)
+        # and we don't need to create any schemas
```

Comment:
👍

Reply:
After moving the calling of unit tests into the test task, this is going to be harder to do differently like this since running unit tests will hook into the standard runnable flow.

Reply:
yeah, exactly. happy to brainstorm alternative approaches in the test task work